src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/parser/subquery/visitor/FindSubqueryInWhereTest.java

```diff
@@ -0,0 +1,110 @@
+
+import com.alibaba.druid.sql.ast.SQLExpr;
+import com.alibaba.druid.sql.ast.expr.SQLExistsExpr;
+import com.alibaba.druid.sql.ast.expr.SQLInSubQueryExpr;
+import com.alibaba.druid.sql.ast.expr.SQLQueryExpr;
+import com.alibaba.druid.sql.parser.ParserException;
+import com.alibaba.druid.sql.parser.Token;
+import com.amazon.opendistroforelasticsearch.sql.parser.ElasticSqlExprParser;
+import com.amazon.opendistroforelasticsearch.sql.parser.subquery.SubqueryType;
+import com.amazon.opendistroforelasticsearch.sql.parser.subquery.visitor.FindSubqueryInWhere;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class FindSubqueryInWhereTest {
+
+    @Test
+    public void findIN() {
+        String sql = "SELECT * FROM A WHERE a IN (SELECT b FROM B)";
+        final FindSubqueryInWhere subqueryInWhere = new FindSubqueryInWhere();
+        parse(sql).accept(subqueryInWhere);
+
+        assertEquals(SubqueryType.IN, subqueryInWhere.subqueryType());
+        assertEquals(1, subqueryInWhere.getSqlInSubQueryExprs().size());
+        assertThat(subqueryInWhere.getSqlInSubQueryExprs().get(0), is(instanceOf(SQLInSubQueryExpr.class)));
+        assertFalse(subqueryInWhere.getSqlInSubQueryExprs().get(0).isNot());
+        assertEquals(0, subqueryInWhere.getSqlExistsExprs().size());
+    }
+
+    @Test
+    public void findNOTIN() {
+        String sql = "SELECT * FROM A WHERE a NOT IN (SELECT b FROM B)";
+        final FindSubqueryInWhere subqueryInWhere = new FindSubqueryInWhere();
+        parse(sql).accept(subqueryInWhere);
+
+        assertEquals(SubqueryType.UNSUPPORTED, subqueryInWhere.subqueryType());
+        assertEquals(1, subqueryInWhere.getSqlInSubQueryExprs().size());
+        assertThat(subqueryInWhere.getSqlInSubQueryExprs().get(0), is(instanceOf(SQLInSubQueryExpr.class)));
+        assertTrue(subqueryInWhere.getSqlInSubQueryExprs().get(0).isNot());
+        assertEquals(0, subqueryInWhere.getSqlExistsExprs().size());
+    }
+
+    @Test
+    public void findEXIST() {
+        String sql = "SELECT * FROM A WHERE EXISTS (SELECT 1 FROM B WHERE A.a_v = B.b_v)";
+        final FindSubqueryInWhere subqueryInWhere = new FindSubqueryInWhere();
+        parse(sql).accept(subqueryInWhere);
+
+        assertEquals(SubqueryType.UNSUPPORTED, subqueryInWhere.subqueryType());
+        assertEquals(1, subqueryInWhere.getSqlExistsExprs().size());
+        assertThat(subqueryInWhere.getSqlExistsExprs().get(0), is(instanceOf(SQLExistsExpr.class)));
+        assertFalse(subqueryInWhere.getSqlExistsExprs().get(0).isNot());
+        assertEquals(0, subqueryInWhere.getSqlInSubQueryExprs().size());
+    }
+
+    @Test
+    public void findNOTEXIST() {
```

Comment:
The naming of tests is inconsistent with next one. I recommend to rename all of them to Java's camelcase (findExist, findNotIn, findInAtSelect)